src/core/core.layouts.js

```diff
@@ -205,6 +205,10 @@ module.exports = {
 
 		helpers.each(leftBoxes.concat(rightBoxes, topBoxes, bottomBoxes), getMinimumBoxSize);
 
+		helpers.each(chartAreaBoxes, function(box) {
+			box.update(maxChartAreaWidth, maxChartAreaHeight);
```

Comment:
`scale.update()` is (unfortunately) very expensive, why do we need to call this method here while we call it already multiple time during the layout? Is the one at the step 9 not enough?

Reply:
I'm trying to take the same approach as the other vertical axis - it sets the `paddingTop` property at the end of `fit` if the axis and ticks are visible, and the step 4 picks it via `getPadding`. For `chartAreaBoxes`, this is the first time for calling `update`.
https://github.com/chartjs/Chart.js/blob/f6d9a39cb8e8661e1d6a733d4b77fb3f3238935a/src/core/core.scale.js#L482-L483

In order to get `paddingTop`, we need to calculate `drawingArea` which is based on the sizes and positions of point labels in the chart.

Reply:
I understand, but I'm very worry about consequences of calling `update()` on scales another time because it's already an issue (#5389). @etimberg any idea?

Reply:
@etimberg Thanks for your advice. I changed the calculation of `largestPossibleRadius` and `yCenter` so that `update()` is called only once.